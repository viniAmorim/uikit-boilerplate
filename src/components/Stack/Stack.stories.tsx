import React from 'react';

import { StackProps } from '@mantine/core';
import { Story, Meta } from '@storybook/react';

import Button from '../Button';
import Stack from './Stack';

export default {
  title: 'Atoms/Display/Stack',
  component: Stack,
} as Meta;

const Template: Story<StackProps> = (args) => <Stack {...args} />;

export const Default = Template.bind({});
Default.args = {
  h: 300,
  children: (
    <>
      <Button variant="secondary">Button 1</Button>
      <Button variant="secondary">Button 2</Button>
      <Button variant="secondary">Button 3</Button>
    </>
  ),
};

export const Center = Template.bind({});
Center.args = {
  h: 300,
  children: (
    <>
      <Button variant="secondary">Button 1</Button>
      <Button variant="secondary">Button 2</Button>
      <Button variant="secondary">Button 3</Button>
    </>
  ),
  align: 'center',
};

export const Space = Template.bind({});
Space.args = {
  h: 300,
  children: (
    <>
      <Button variant="secondary">Button 1</Button>
      <Button variant="secondary">Button 2</Button>
      <Button variant="secondary">Button 3</Button>
    </>
  ),
  justify: 'space-between',
};
