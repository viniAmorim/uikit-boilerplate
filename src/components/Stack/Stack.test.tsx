import React from 'react';

import { render } from '@testing-library/react';

import Stack from './Stack';

describe('Stack', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Stack>
        <div>1</div>
        <div>2</div>
      </Stack>,
    );

    const stack = container.firstChild;

    expect(stack).toBeInTheDocument();
    expect(stack).toMatchSnapshot();
    expect(stack).toHaveStyle(`
    display: flex;
    flex-direction: column;
  `);
  });
  it('should render children with correct flexbox properties', () => {
    const { container } = render(
      <Stack align="flex-start" justify="flex-start" spacing={3}>
        <div>1</div>
        <div>2</div>
      </Stack>,
    );
    const stack = container.firstChild;
    expect(stack).toHaveStyle(`
      justify-content: flex-start;
      align-items: flex-start;
      gap: 0.1875rem;
    `);
  });
});
