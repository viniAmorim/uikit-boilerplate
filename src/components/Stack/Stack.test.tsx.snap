// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack should render correctly 1`] = `
<div
  class="mantine-Stack-root mantine-caxjnw"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;
